bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/AbstractAsyncBundleProcessor.java

```diff
@@ -107,16 +107,16 @@ protected boolean isBundleRelevant(Bundle bundle) {
     }
 
     /**
-     * Checks for the existence of a given directors inside the bundle.
+     * Checks for the existence of a given directors inside the bundle and its attached fragments.
      *
      * Helper method which can be used in {@link #isBundleRelevant(Bundle)}.
      * 
      * @param bundle
      * @param path the directory name to look for
-     * @return <code>true</code> id the bundle contains the given directory
+     * @return <code>true</code> id the bundle or one of its attached fragments contain the given directory
      */
     protected final boolean isDirectoryPresent(Bundle bundle, String path) {
-        return bundle.getEntry(path) != null;
+        return bundle.findEntries(path, "*", true) != null;
```

Comment:
Ok, these are valid points. We could set the third parameter to false as it is really not necessary to do this recursive. We only want to check the existence of the *path*, so I could change the code to 
```java 
return bundle.findEntries(path, null, false) != null;
``` 
However, this is still listing each file *within* the path, and for ESH-INF/config this could be some. I wonder, if there is anywhere deep in the OSGi API another chance to check for the directory existence.